src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -288,7 +288,10 @@ public KeyRangeIterator.Builder getIndexQueryResults(Collection<Expression> expr
 
     private void maybeTriggerGuardrails(QueryViewBuilder.QueryView queryView)
     {
-        int referencedIndexes = queryView.referencedIndexes.size();
+        int referencedIndexes = 0;
+
+        for (Pair<Expression, Collection<SSTableIndex>> expressionSSTables : queryView.view)
+            referencedIndexes = Math.max(referencedIndexes, expressionSSTables.right.size());
```

Comment:
This guardrail was designed to make sure individual column predicates don't involve index queries that hit too many SSTables rather than worrying about the total number of SSTables across all column predicates. The Harry tests can have 10+ column predicates, and that makes it much more likely we'll hit this and give in-JVM tests trouble.